scrape/scrape.go

```diff
@@ -509,6 +514,21 @@ func (sp *scrapePool) sync(targets []*Target) {
 		hash := t.hash()
 
 		if _, ok := sp.activeTargets[hash]; !ok {
+			// The scrape interval and timeout labels are set to the config's values initially,
+			// so whether changed via relabeling or not, they'll exist and hold the correct values
+			// for every target.
+			intervalLabel := t.labels.Get(model.ScrapeIntervalLabel)
+			interval, err := time.ParseDuration(intervalLabel)
+			if err != nil {
+				level.Error(sp.logger).Log("msg", "Error parsing interval label", "err", err, "value", intervalLabel)
+				continue
+			}
+			timeoutLabel := t.labels.Get(model.ScrapeTimeoutLabel)
+			timeout, err := time.ParseDuration(timeoutLabel)
+			if err != nil {
+				level.Error(sp.logger).Log("msg", "Error parsing timeout label", "err", err, "value", timeoutLabel)
+			}
```

Comment:
We should check that the timeout <= scrape interval; here is e.g. what we have now:

parsing YAML file prometheus.yml: global scrape timeout greater than scrape interval

Reply:
I'll add that in the `populateLabels` function in `target.go`, which is where all the label checking is done. They're only parsed here again because there's nowhere to store the parsed value in the intermediate.

Reply:
Theoretically, there should never be an error here.